tests/test_cnes_sonar_scanner.py

```diff
@@ -0,0 +1,517 @@
+        """
+        if activate_rule:
+            # Get the key of the Quality Profile to use
+            qp_key = requests.get(f"{cls.SONARQUBE_LOCAL_URL}/api/qualityprofiles/search?quality_profile={quality_profile}",
+                auth=("admin", cls.SONARQUBE_ADMIN_PASSWORD)).json()['profiles'][0]['key']
+            # Activate the rule in the Quality Profile to allow the Sensor to be used
+            requests.post(f"{cls.SONARQUBE_LOCAL_URL}/api/qualityprofiles/activate_rule",
+                auth=("admin", cls.SONARQUBE_ADMIN_PASSWORD),
+                data={
+                    "key": qp_key,
+                    "rule": rule_violated
+                })
+        # Create a project on SonarQube
+        errors = requests.post(f"{cls.SONARQUBE_LOCAL_URL}/api/projects/create",
+            auth=("admin", cls.SONARQUBE_ADMIN_PASSWORD),
+            data={
+                "name": project_name,
+                "project": project_key
+            }).json().get('errors', [])
+        assert not errors
+        # Set its Quality Profile for the given language to the given one
+        requests.post(f"{cls.SONARQUBE_LOCAL_URL}/api/qualityprofiles/add_project",
+            auth=("admin", cls.SONARQUBE_ADMIN_PASSWORD),
+            data={
+                "language": language_key,
+                "project": project_key,
+                "qualityProfile": quality_profile
+            })
+        # Analyse the project and collect the analysis files (that match the default names)
+        docker_client = docker.from_env()
+        analysis_output = docker_client.containers.run(cls._SONAR_SCANNER_IMAGE,
+            f"-Dsonar.projectKey={project_key} -Dsonar.projectName=\"{project_name}\" -Dsonar.projectVersion=1.0 -Dsonar.sources={source_folder}",
+            auto_remove=True,
+            user=f"{os.getuid()}:{os.getgid()}",
+            volumes={
+                f"{cls._PROJECT_ROOT_DIR}": {'bind': '/usr/src', 'mode': 'rw'},
+                f"{cls._PROJECT_ROOT_DIR}/.sonarcache": {'bind': '/opt/sonar-scanner/.sonar/cache', 'mode': 'rw'}
+            },
+            environment={"SONAR_HOST_URL": cls.SONARQUBE_URL},
+            network=cls.SONARQUBE_NETWORK,
+            working_dir=f"/usr/src/{language_folder}").decode("utf-8")
+        for line in (expected_sensor, expected_import):
+            # Hint: if this test fails, the sensor for the tool or for the importation was not launched
+            assert line in analysis_output
+        # Wait for SonarQube to process the results
+        time.sleep(10)
+        # Check that the issue was added to the project
+        issues = requests.get(f"{cls.SONARQUBE_LOCAL_URL}/api/issues/search?componentKeys={project_key}",
+            auth=("admin", cls.SONARQUBE_ADMIN_PASSWORD)).json()['issues']
+        nb_issues = len([ issue for issue in issues if issue['rule'] == rule_violated ])
+        # Hint: an issue must be raised by the rule violated
+        assert nb_issues == 1
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/gh/cnescatlab/sonar-scanner/pullRequest?prid=6252273)